autoload/neomake/makers/ft/haskell.vim

```diff
@@ -1,55 +1,61 @@
 unlet! s:makers
-
-function! s:MakerAvailable(command) abort
-    " stack may be able to find a maker binary that's not on the normal path
-    " so check for that first
-    if executable('stack')
-        " run the maker command using stack to see whether stack can find it
-        " use the help flag to run the maker command without doing anything
-        if system('stack exec -- ' . a:command . ' --help > /dev/null 2>&1; echo $?') == 0
-            return 1
-        else " if stack cannot find the maker command, its not available anywhere
-            return 0
-        endif
-    elseif executable(a:command) " stack isn't available, so check for the maker binary directly
-        return 1
-    else
-        return 0
-    endif
-endfunction
+unlet! s:uses_cabal
 
 function! neomake#makers#ft#haskell#EnabledMakers() abort
-    " cache whether each maker is available, to avoid lots of (UI blocking) system calls...the user must restart vim if a maker's availability changes
     if !exists('s:makers')
+        " cache whether each maker is available, to avoid lots of (UI blocking)
+        " system calls
+        " TODO: figure out how to do all this configuration async instead of
+        " caching it--that would allow the user to change directories from
+        " within vim and recalculate maker availability without restarting vim
         let commands = ['ghc-mod', 'hdevtools', 'hlint', 'liquid']
         let s:makers = []
+        let s:jobs = []
         for command in commands
-            if s:MakerAvailable(command)
+            " stack may be able to find a maker binary that's not on the normal
+            " path so check for that first
+            if executable('stack')
+                " run the maker command using stack to see whether stack can
+                " find it use the help flag to run the maker command without
+                " doing anything
+                let stack_command = [
+                      \   'stack'
+                      \ , 'exec'
+                      \ , '--'
+                      \ , command
+                      \ , '--help'
+                      \ ]
+                if has('nvim')
+                    let job_id = jobstart(
+                        \ stack_command,
+                        \ { 'command': command
+                        \ , 'on_exit': function('s:CheckStackMakerAsync')
+                        \ })
+                    if job_id > 0
+                        call add(s:jobs, job_id)
+                    endif
+                else
+                    extend(stack_command, '> /dev/null 2>&1;', 'echo $?')
```

Comment:
s/extend/call extend/

Too bad that apparently only the linter caught this.. :/